core/trino-main/src/main/java/io/trino/execution/AddColumnTask.java

```diff
@@ -142,7 +146,7 @@ public ListenableFuture<Void> execute(
 
             ColumnMetadata column = ColumnMetadata.builder()
                     .setName(columnName.getValue())
-                    .setType(type)
+                    .setType(getSupportedType(session, catalogHandle, tableMetadata.metadata().getProperties(), type))
```

Comment:
Can we use `ConnectorMetadata#getTableProperties` instead of fetching `ConnectorTableMetadata` ?

Reply:
`ConnectorMetadata#getTableProperties` returns `ConnectorTableProperties`. It's different from `Map<String, Object> properties` in `ConnectorTableMetadata`.